OpenRA.Mods.Common/Widgets/Logic/Editor/ActorEditLogic.cs

```diff
@@ -0,0 +1,241 @@
+			deleteButton.OnClick = () => Delete();
+			actorEditPanel.IsVisible = () => actorSelectBorder.IsVisible();
+			actorSelectBorder.IsVisible = () => currentActor != null
+				&& editor.CurrentBrush == editor.DefaultBrush
+				&& Game.RunTime > lastScrollTime + scrollVisibleTimeout;
+
+			actorIdField.OnEscKey = () =>
+			{
+				actorIdField.YieldKeyboardFocus();
+				return true;
+			};
+
+			// Setup owners drop down
+			selectedOwner = editorActorLayer.Players.Players.Values.First();
+			Func<PlayerReference, ScrollItemWidget, ScrollItemWidget> setupItem = (option, template) =>
+			{
+				var item = ScrollItemWidget.Setup(template, () => selectedOwner == option, () =>
+				{
+					ownersDropDown.Text = option.Name;
+					ownersDropDown.TextColor = option.Color.RGB;
+					selectedOwner = option;
+				});
+
+				item.Get<LabelWidget>("LABEL").GetText = () => option.Name;
+				item.GetColor = () => option.Color.RGB;
+				return item;
+			};
+
+			ownersDropDown.OnClick = () =>
+			{
+				var owners = editorActorLayer.Players.Players.Values.OrderBy(p => p.Name);
+				ownersDropDown.ShowDropDown("LABEL_DROPDOWN_TEMPLATE", 270, owners, setupItem);
+			};
+
+			ownersDropDown.Text = selectedOwner.Name;
+			ownersDropDown.TextColor = selectedOwner.Color.RGB;
+		}
+
+		public override void Tick()
+		{
+			var actor = editor.DefaultBrush.SelectedActor;
+			if (actor != null)
+			{
+				var origin = worldRenderer.Viewport.WorldToViewPx(new int2(actor.Bounds.X, actor.Bounds.Y));
+
+				// If we scrolled, hide the edit box for a moment
+				if (actorSelectBorder.Bounds.X != origin.X || actorSelectBorder.Bounds.Y != origin.Y)
+				{
+					lastScrollTime = Game.RunTime;
+				}
+
+				// If we changed actor, move widgets
+				if (currentActor != actor)
+				{
+					lastScrollTime = 0; // Ensure visible
+					selectedOwner = editorActorLayer.Players.Players.Values.First(x => x.Name == actor.Owner.Name);
+					ownersDropDown.Text = selectedOwner.Name;
+					ownersDropDown.TextColor = selectedOwner.Color.RGB;
+
+					if (currentActor != null) // Hide border of the previously selected actor.
+						currentActor.ShowBorder = false;
+					currentActor = actor;
+					currentActor.ShowBorder = true;
+
+					actorIdField.Text = actor.ID;
+					typeLabel.Text = actor.TypeSummary;
+					actorIdField.CursorPosition = actor.ID.Length;
+					actorSelectBorder.Bounds.Width = actor.Bounds.Width * 2;
+					actorSelectBorder.Bounds.Height = actor.Bounds.Height * 2;
+					HideValidationMessage();
+				}
+
+				actorSelectBorder.Bounds.X = origin.X;
+				actorSelectBorder.Bounds.Y = origin.Y;
+
+				// Set the edit panel to the right of the selection border.
+				int newX = origin.X + actorSelectBorder.Bounds.Width + editPanelPadding;
+				actorEditPanel.Bounds.X = newX;
+				actorEditPanel.Bounds.Y = origin.Y;
+			}
+			else
+			{
+				// Selected actor is null, hide the border and edit panel.
+				if (currentActor != null)
+					currentActor.ShowBorder = false;
```

Comment:
All this manual manipulation of `ShowBorder` / `Selected` is asking for trouble.  I suggest making `currentActor` a property, and defining a setter that does this automatically when the value is changed.